ls1intum/Artemis
# `General`: Fix breadcrumb routing from teams page + add teams button to programming exercise details

### Checklist
#### General


#### Client

### Motivation and Context

Fixes #4653.

### Description


- Modified the breadcrumb generation code to fetch the exercise by id and include the type in the breadcrumb URI for backlinks to exercise detail pages
- Added Teams button to the programming exercise buttons on the detail page if the exercise is in team mode

### Steps for Testing

Prerequisites:
- 1 Programming Exercise in Team Mode
- 1 Instructor

1. Navigate to the Details Page of the Exercise. The Teams button should be shown in the second line of buttons.
2. Click the button. You should be redirected to the teams list.
3. Click the breadcrumb link with the name of the exercise. The link should point to .../programming-exercises/... instead of .../exercises/... and forward you to the exercise details page again.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots


![grafik](https://user-images.githubusercontent.com/23171488/161262050-f7ca3e83-4288-4f1a-9a17-918b1b28adf1.png)


https://user-images.githubusercontent.com/23171488/161262213-6b7ae436-a148-4f0e-8bd0-5f587cdb907e.mov
